code/frameworks/svelte-vite/src/preset.ts

```diff
@@ -24,7 +24,7 @@ export const viteFinal: StorybookConfig['viteFinal'] = async (config, { presets
 const removeSvelteKitPlugin = (plugins: PluginOption[]) => {
   plugins.forEach((plugin, index) => {
     if (plugin && 'name' in plugin && plugin.name === 'vite-plugin-svelte-kit') {
-      // eslint-disable-next-line no-param-reassign -- we explicitly want to mutate the array as stated in Vite docs
+      // eslint-disable-next-line no-param-reassign -- we explicitly want to mutate the array as stated here: https://vitejs.dev/guide/api-plugin.html#config
```

Comment:
oh yeah, I hadn't even noticed that.
I have a version that is more declarative, but I'm unsure if it's actually more readable. But I don't have any strong opinions so if you think it's better I'll happily push it.

```ts
export const viteFinal: StorybookConfig['viteFinal'] = async (config, { presets }) => {
  const { plugins = [] } = config;

  return {
    ...config,
    plugins: [
      // remove SvelteKit plugin
      ...plugins.map(withoutSvelteKitPlugin),
      // Add svelte plugin if not present
      !hasPlugin(plugins, 'vite-plugin-svelte') &&
        (await import('@sveltejs/vite-plugin-svelte')).svelte(),
      // Add docgen plugin
      svelteDocgen(config),
    ].filter(Boolean),
  };
};

const withoutSvelteKitPlugin = (plugin: PluginOption): PluginOption => {
  if (Array.isArray(plugin)) {
    // recursive - Vite plugins can be nested
    return plugin.map(withoutSvelteKitPlugin).filter(Boolean);
  }
  if (plugin && 'name' in plugin && plugin.name === 'vite-plugin-svelte-kit') {
    return undefined;
  }
  return plugin;
};
```

Reply:
Thanks, the way you've got it right now is fine I think, and hopefully this is temporary anyway, until we can figure out what's causing the app to load instead of storybook in 7.0.